autometa/common/external/diamond.py

```diff
@@ -343,35 +504,80 @@ def add_taxids(hits, database, verbose=True):
     if not os.path.exists(database):
         raise FileNotFoundError(database)
     # OPTIMIZE: prot.accession2taxid.gz database is almost 1 billion lines...
-    accessions = set(chain.from_iterable([hit.sseqids.keys() for qseqid,hit in hits.items()]))
-    fh = gzip.open(database) if database.endswith('.gz') else open(database)
-    __ = fh.readline()
+    accessions = set(
+        chain.from_iterable([hit.sseqids.keys() for qseqid, hit in hits.items()])
+    )
+    # "rt" open the database in text mode instead of binary. Now it can be handled like a text file
+    fh = gzip.open(database, "rt") if database.endswith(".gz") else open(database)
+    __ = fh.readline()  # remove the first line as it just gives the description
     if verbose:
-        logger.debug(f'Searching for {len(accessions):,} accessions in {os.path.basename(database)}. This may take a while...')
-    is_gzipped = True if database.endswith('.gz') else False
+        logger.debug(
+            f"Searching for {len(accessions):,} accessions in {os.path.basename(database)}. This may take a while..."
+        )
     n_lines = file_length(database) if verbose else None
-    desc = f'Parsing {os.path.basename(database)}'
+    desc = f"Parsing {os.path.basename(database)}"
     acc2taxids = {}
     for line in tqdm(fh, disable=disable, desc=desc, total=n_lines, leave=False):
-        line = line.decode() if is_gzipped else line
-        acc_num, acc_ver, taxid, _ = line.split('\t')
+        acc_num, acc_ver, taxid, _ = line.split("\t")
         taxid = int(taxid)
         if acc_num in accessions:
-            acc2taxids.update({acc_num:taxid})
+            acc2taxids.update({acc_num: taxid})
         if acc_ver in accessions:
-            acc2taxids.update({acc_ver:taxid})
+            acc2taxids.update({acc_ver: taxid})
     fh.close()
     n_qseqids = len(hits)
-    desc = f'Translating {n_qseqids} qseqids\' accessions to taxids'
-    for qseqid,hit in tqdm(hits.items(), disable=disable, total=n_qseqids, desc=desc, leave=False):
+    desc = f"Translating {n_qseqids} qseqids' accessions to taxids"
+    for qseqid, hit in tqdm(
+        hits.items(), disable=disable, total=n_qseqids, desc=desc, leave=False
+    ):
         for sseqid in hit.sseqids:
             taxid = acc2taxids.get(sseqid)
-            # if taxid is None:
-            #     raise DatabasesOutOfDateError(f'{sseqid} deprecated/suppressed/removed')
-            hit.sseqids[sseqid].update({'taxid':taxid})
+            hit.sseqids[sseqid].update({"taxid": taxid})
     return hits
 
-def main(args):
+
+def main():
+    import argparse
+    import os
+
+    parser = argparse.ArgumentParser(
+        description="""
+    Retrieves blastp hits with provided input assembly
+    """
+    )
```

Comment:
Since you are specifying multiple defaults in the argparse parser. You should apply the defaults formatter.

https://github.com/KwanLab/Autometa/blob/ddc2bf951d6c4a8c29e9e2301e15a762327b0267/docs/template.py#L89-L94

```suggestion
    parser = argparse.ArgumentParser(
        description="Retrieves blastp hits with provided input assembly",
        formatter_class=argparse.ArgumentDefaultsHelpFormatter
    )
```